core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionServer.scala

```diff
@@ -268,6 +270,31 @@ private[rest] abstract class ClearRequestServlet extends RestServlet {
   protected def handleClear(): ClearResponse
 }
 
+/**
+ * A servlet for handling readyz requests passed to the [[RestSubmissionServer]].
+ */
+private[rest] abstract class ReadyzRequestServlet extends RestServlet {
+
+  /**
+   * Return the status of master is ready or not.
+   */
+  protected override def doGet(
+      request: HttpServletRequest,
+      response: HttpServletResponse): Unit = {
+    val readyzResponse = handleReadyz()
+    val responseMessage = if (readyzResponse.success) {
+      response.setStatus(HttpServletResponse.SC_OK)
+      readyzResponse
+    } else {
+      response.setStatus(HttpServletResponse.SC_SERVICE_UNAVAILABLE)
+      handleError("Master is not ready.")
```

Comment:
Hmm, so if the master is not ready, we treat it as an error (i.e., usually `handleError` signals that an exception has been thrown)? Or we should return a `ReadyzResponse` with success set to false?

Reply:
No, that behavior is defined by K8s in the PR description (https://kubernetes.io/docs/reference/using-api/health-checks/)
> so if the master is not ready, we treat it as an error (i.e., usually handleError signals that an exception has been thrown)?

In this context, the readiness is `non-STANDBY` mode because of the existing workaround which is used in K8s.
> But it is not ready doesn't mean it is an error, right?

Reply:
Mostly, it's determined by HTTPS status. In this case, 503, instead of the content of message.
```
$ curl -vv http://localhost:6066/v1/submissions/readyz
...
< HTTP/1.1 503 Service Unavailable
```